src/kili/use_cases/question/question_use_case.py

```diff
@@ -0,0 +1,61 @@
+"""Question use cases."""
+
+from dataclasses import dataclass
+from typing import List, Optional
+
+from kili.adapters.kili_api_gateway.issue.types import IssueToCreateKiliAPIGatewayInput
+from kili.domain.asset.asset import AssetExternalId, AssetId
+from kili.domain.project import ProjectId
+from kili.domain.question import QuestionId
+from kili.use_cases.base import BaseUseCases
+from kili.use_cases.utils import UseCasesUtils
+
+
+@dataclass
+class QuestionToCreateUseCaseInput:
+    """Question to create use case input."""
+
+    text: Optional[str]
+    asset_id: Optional[str]
+    asset_external_id: Optional[str]
+
+
+class QuestionUseCases(BaseUseCases):
+    """Question use cases."""
+
+    def create_questions(
+        self,
+        project_id: ProjectId,
+        questions: List[QuestionToCreateUseCaseInput],
+    ) -> List[QuestionId]:
+        """Create questions."""
+        if questions[0].asset_id is not None:
+            # we assume that if 1 question asset Id is not None, all there others are too
+            asset_id_array = [AssetId(question.asset_id) for question in questions]  # type: ignore
+            external_id_array = None
+        else:
+            asset_id_array = None
+            external_id_array = [
+                AssetExternalId(question.asset_external_id)  # type: ignore
```

Comment:
better doing runtime checks with to make sure it's really the case

and we should avoid type ignore

Reply:
Iterating one more time over possibly 25 000 (and soon 100.000) assets to find out if the ids are defined or not is costly. If there are mixed values asset_ids and external_ids, then the server will end up throwing an error so the DX should be acceptable. Added the casting.